src/test/java/org/example/FilteringAndSearchingTest.java

```diff
@@ -0,0 +1,207 @@
+        selectBrand(brandName);
+
+        WebElement searchDropdownList = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.visibilityOfElementLocated(By.id("a-autoid-0-announce")));
+
+        searchDropdownList.click();
+
+        WebElement selectDropdownList = webDriver.findElement(By.id("s-result-sort-select_1"));
+        selectDropdownList.click();
+
+        boolean arePricesInAscendingOrder = verifyPricesAreInAscendingOrder();
+
+        Assert.assertTrue(arePricesInAscendingOrder, "Prices are not in ascending order");
+    }
+
+    private void selectBrand(String brandName){
+        WebElement brandNameCheckbox = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                .until(ExpectedConditions.presenceOfElementLocated(By.xpath("//span[text() = '" + brandName + "']")));
+        brandNameCheckbox.click();
+    }
+
+
+    private boolean verifyAnyTitleContainsBrandName(String brandName) {
+        boolean anyTitleContainsInputWord = false;
+
+        int numOfPages = webDriver.findElements(By.xpath("//span[@class='s-pagination-strip']/span[4]")).isEmpty() ?
+                1 : Integer.parseInt(webDriver.findElement(By.xpath("//span[@class='s-pagination-strip']/span[4]")).getText());
+
+        for (int i = 1; i <= numOfPages; i++) {
+//checking if every title of the page contains brand name
+            anyTitleContainsInputWord = webDriver.findElements(By.xpath("//div[contains(@class,'s-card-container')]//span[contains(@class,'a-size-medium')]"))
+                    .stream()
+                    .map(WebElement::getText)
+                    .map(String::toLowerCase)
+                    .allMatch(e -> e.contains(brandName.toLowerCase()));
+
+//if any title doesnt contain brand name, break the loop
+            if (!anyTitleContainsInputWord) {
+                break;
+            }
+
+//click next button until its not clickable, based on num of pages
+            if (numOfPages != i) {
+                WebElement nextButton = new WebDriverWait(webDriver, Duration.ofSeconds(6))
+                        .until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//*[contains(@class,'s-pagination-next')]")));
+                nextButton.click();
+            }
+        }
+        return anyTitleContainsInputWord;
+    }
+
+    private boolean verifyPricesAreInChoseRange(float minPrice, float maxPrice) {
+        boolean arePricesInChosenRange = false;
+
+        int numOfPages = webDriver.findElements(By.xpath("//span[@class='s-pagination-strip']/span[4]")).isEmpty() ?
+                1 : Integer.parseInt(webDriver.findElement(By.xpath("//span[@class='s-pagination-strip']/span[4]")).getText());
+
+        for (int i = 1; i <= numOfPages; i++) {
+//checking if prices are within specified range
+            arePricesInChosenRange = webDriver.findElements(By.xpath("//span[@class=\"a-price\"]/span[@class=\"a-offscreen\"]"))
```

Comment:
Your xpath finds elements from outside of results list which gives me false negative test result.

![image](https://user-images.githubusercontent.com/100229236/186028464-436dd14f-f368-4a77-ae42-06702dcc2f74.png)

Reply:
Oh sorry, I didn't see your previous comment, that you just found what is wrong ;) I hope that you will deal with this problem.